src/Appwrite/Utopia/Response.php

```diff
@@ -491,44 +491,45 @@ public function output(Document $document, string $model): array
                 }
             }
 
-            if ($rule['array']) {
-                if (!is_array($document[$key])) {
-                    throw new Exception($key . ' must be an array of type ' . $rule['type']);
-                }
+            if ($data->isSet($key)) {
+                if ($rule['array']) {
+                    if (!is_array($document[$key])) {
+                        throw new Exception($key . ' must be an array of type ' . $rule['type']);
+                    }
```

Comment:
Do we need this change?

Reply:
Yes there's a bug, 
We only wanna do this if the key exists, because if it's not required attribute and it doesn't exist, this code produces bug

Reply:
Okay the error is when the attribute is an array and it is unset. We are throwing exception if the type is not array when rule says it is array, it's not array because it's optional and not set. 

we should throw exception when it's array and the value is not array. 
right now value is not set so it's throwing exception when it should not

this is the exception "orders must be an array of type string"

Reply:
Is this caused by using a select query but not selecting `orders`? I think the problem will be larger than this. For any resource, if we use select queries and don't select any attributes that are required, this function will set them to their default values even though they should not be included in the response at all.

I think we should block select queries for all resources _except_ documents. We can do this by removing `new Select($attributes)` from `Base`. We can then revert the changes here as well as the `$prohibitedQueries` in `Base`

Reply:
I have removed the Select validator and the prohibitQueries from the Base validator.

I would still like to keep `$data->isSet($key)` because this seems like a reasonable condition for the code to work. 

No sense in throwing an exception that the rule is not an array if it's unset in the response since it's not a required rule

Reply:
If we block the select queries for indexes (and all other resources), I don't think the exception would be thrown. In the database, `orders` always has a value set, because when not included in the request to create an index, the default of an empty array is set. So the only way that exception could be thrown, is if using a select query and not selecting `orders`.

Reply:
Hmm, that makes sense. But this condition is also reasonable. I think we should let this stay even if this case might not happen for us because, logically, this code block should run on set keys only, and we should check if the key is set or not in the output if the rule is an array.

Reply:
Removed this condition.